backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java

```diff
@@ -350,22 +350,32 @@ public RepositoryEntity getProtectedRepositoryById(
 
   private Optional<RepositoryEntity> getRepositoryByName(
       Session session, String name, Workspace workspace, RepositoryTypeEnum repositoryType) {
-    List<Long> repositoryIds = getRepositoryEntityIdsByName(session, name, repositoryType);
+    ModelDBServiceResourceTypes modelDBServiceResourceTypes =
+        ModelDBServiceResourceTypes.REPOSITORY;
+    if (repositoryType.equals(RepositoryTypeEnum.DATASET)) {
+      modelDBServiceResourceTypes = ModelDBServiceResourceTypes.DATASET;
+    }
     List<GetResourcesResponseItem> accessibleAllWorkspaceItems =
         roleService.getResourceItems(
-            workspace,
-            !repositoryIds.isEmpty()
-                ? repositoryIds.stream().map(String::valueOf).collect(Collectors.toSet())
-                : Collections.emptySet(),
-            ModelDBServiceResourceTypes.REPOSITORY);
-    Optional<Long> repoId =
+            workspace, Collections.emptySet(), modelDBServiceResourceTypes);
+    Set<Long> repoIds =
         accessibleAllWorkspaceItems.stream()
+            .filter(
+                getResourcesResponseItem -> getResourcesResponseItem.getResourceName().equals(name))
```

Comment:
If you do an RPC and then do a filter, it's wasteful. Can the filter be pushed to the remote system?

Reply:
Yes but still remote system (UAC) do not have any endpoint which is give me a filter data based on the name.